js/BranchingSet.js

```diff
@@ -56,41 +121,59 @@ export default class BranchingSet {
 
   getNextModel({ isTheoretical = false } = {}) {
     const config = this.model.get('_branching');
-    const brachingModels = this.models;
+    const branchingModels = this.models;
     const branchedModels = this.branchedModels;
 
     const isBeforeStart = !branchedModels.length;
     if (isBeforeStart) {
       const hasStartId = Boolean(config._start);
       const firstModel = hasStartId ?
-        brachingModels.find(model => model.get('_id') === config._start) :
-        brachingModels[0];
+        branchingModels.find(model => model.get('_id') === config._start) :
+        branchingModels[0];
       return firstModel;
     }
 
     const lastChildModel = branchedModels[branchedModels.length - 1];
-    const isLastIncomplete = !lastChildModel.get('_isComplete');
+    const isLastIncomplete = !lastChildModel.get('_isComplete') && !lastChildModel.get('_isOptional');
     if (isLastIncomplete && !isTheoretical) {
       return false;
     }
 
     const lastChildConfig = lastChildModel.get('_branching');
-    if (!lastChildConfig || !lastChildConfig._isEnabled) return true;
+    if (!lastChildConfig) return true;
 
     // Branch from the last model's correctness, if configured
     const correctness = getCorrectness(lastChildModel);
     const nextId = lastChildConfig._force || lastChildConfig[`_${correctness}`];
     if (!nextId) return true;
 
-    const isRelativeId = nextId.includes('@');
-    if (!isRelativeId) {
-      return brachingModels.find(model => model.get('_id') === nextId) || true;
+    function findNextModel(nextId) {
+      const isRelativeId = nextId.includes('@');
+      if (!isRelativeId) {
+        return branchingModels.find(model => model.get('_id') === nextId);
+      }
+      const originalLastChildModel = data.findById(lastChildModel.get('_branchOriginalModelId'));
+      const nextModel = originalLastChildModel.findRelativeModel(nextId);
+      const wasModelAlreadyUsed = nextModel.get('_isAvailable');
+      if (wasModelAlreadyUsed) return true;
+      return nextModel;
     }
 
-    const originalLastChildModel = data.findById(lastChildModel.get('_branchOriginalModelId'));
-    const nextModel = originalLastChildModel.findRelativeModel(nextId);
-    const wasModelAlreadyUsed = nextModel.get('_isAvailable');
-    if (wasModelAlreadyUsed) return true;
+    const nextModel = findNextModel(nextId);
+    // Mark as finished
+    if (nextModel === true) return true;
+    if (nextModel === undefined) {
+      // Set the start block of a different branching set
+      const nextBranchingSet = Adapt.branching.getSubsetByModelId(nextId);
+      if (!nextBranchingSet) {
+        logging.error(`Cannot branch to a model that isn't contained in a branching set: ${nextId} from ${lastChildModel.get('_id')}`);
```

Comment:
We had previously spoken about changing this to warn, to prevent previous configurations from breaking, when an id had been configured on the last child.